### ISSUE TITLE
[Maintenance] Fix remaining lint issues

### ISSUE THREAD

**Issue by yozlet** [18F/web-design-standards#1190]
## Issue type

Maintenance, Code Quality
## Description
- [x] @rogeruiz & I fixed a load of `scss-lint` warnings in #1180 but some remain - see [Roger's comment](https://github.com/18F/web-design-standards/pull/1180#issuecomment-221627503)
  
  > **Previously:** We've been wondering whether to either swap out `scss-lint` for a more modern linter or tweak some of the rules (such as turning off `PropertyCount` or raising its limit). We should have that discussion first.
- [x] After replacing `scss-lint` with [stylelint](http://stylelint.io/) in #1419, we now have a ton of linting errors to deal with before we can re-enable the `stylelint` task a prerequisite for the `build` gulp task and the `test` npm script.

**Comment by msecret** [comment]
@yozlet I've been working with Adam on using Stylelint for ndoe-based linting across 18F, so I'd hold off on trying to find a new linting tool

We've discussed changing some of the rules, such as the selector depth. I think we should evaluate where we're breaking the rules and attempt to refactor code before we change any rules.

**Comment by joshbruce** [comment]
To throw some copper on the matter. I do find not being able to go 3 deep as being problematic in some areas - particularly with more compound navigation/menus.

```
.primary-navigation {
  ul {
    li {
      ul {
...
      }
    }
  }
}
```

To comply with the linter, I believe, classes would need to be added; or some other solution that hasn't come to me yet.

**Comment by shawnbot** [comment]
FYI, I'm working on incorporating the stylelint work that y'all have done into the split repo work that @juliaelman started. Mostly I've just nixed all of the `scss-lint` references and replaced them with  `stylelint`, but one important thing to note is that I've had to disable the `stylelint` prerequisite to the `sass` gulp task because it fails (there are a _lot_ of linting errors). I've also removed the linting task from the `test` script so that the linting errors don't fail the CI build.

I think it's best to push off fixing the linting errors to another issue and reincorporate the linting steps into the build and test scripts once they're working again. Thoughts, anyone?

**Comment by shawnbot** [comment]
I think it's actually worth either re-opening this issue or creating a new one. Any preferences?

**Comment by shawnbot** [comment]
Okay, reopening!

**Comment by juliaelman** [comment]
Added stylelint here and closing this issue: https://github.com/18F/web-design-standards/pull/1419
